core/types/celo_denominated_tx.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+)
+
+var (
+	ErrNonCeloDenominated error = errors.New("Tx not CeloDenominated")
+)
+
+type CeloDenominatedTx struct {
+	ChainID             *big.Int
+	Nonce               uint64
+	GasTipCap           *big.Int
+	GasFeeCap           *big.Int
+	Gas                 uint64
+	FeeCurrency         *common.Address `rlp:"nil"` // nil means native currency
+	To                  *common.Address `rlp:"nil"` // nil means contract creation
+	Value               *big.Int
+	Data                []byte
+	AccessList          AccessList
+	MaxFeeInFeeCurrency *big.Int
+
+	// Signature values
+	V *big.Int `json:"v" gencodec:"required"`
+	R *big.Int `json:"r" gencodec:"required"`
+	S *big.Int `json:"s" gencodec:"required"`
+}
+
+// copy creates a deep copy of the transaction data and initializes all fields.
+func (tx *CeloDenominatedTx) copy() TxData {
+	cpy := &CeloDenominatedTx{
```

Comment:
Why is `MaxFeeInFeeCurrency` not copied?

Reply:
same mistake as previous